### DISCUSSION TITLE
EdgeX TAF testing using docker on RISC_V

### DISCUSSION THREAD

**Discussion by neerajasjawali** [edgexfoundry/community#120] [Q&A]
Hi All,

I am working on EdgeX TAF (jakarta version) analysis with security services using docker on Risc-V platform.
1) Within the TAF shell script, the architectures defined are x86 and arm. While testing it on risc-v system using the below command i am facing an error
command: `sh run-tests.sh riscv64 false functional-test core-metadata `
error: 
```
root@NB2-SOC-BSP-02FEB23-REL12-v1:~/edgex-taf/TAF/utils/scripts/docker# sh run-tests.sh riscv64 false functional-test core-metadata
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 20609  100 20609    0     0   176k      0 --:--:-- --:--:-- --:--:--  178k
{"msg":"exec the container process: Exec format error","level":"error","time":"2023-05-31T10:23:51.000130608Z"}
{"msg":"exec the container process: Exec format error","level":"error","time":"2023-05-31T10:23:52.000885007Z"}
{"msg":"exec the container process: Exec format error","level":"error","time":"2023-05-31T10:23:56.000651199Z"}

```

2) Within the run-tests.sh shell script, do I need to change the below variables for risc-v?

```
 # x86_64 or arm64
[ "$USE_ARCH" = "arm64" ] && USE_ARM64="-arm64"

# Common Variables
USE_SHA1=jakarta  # edgex-compose branch or SHA1
TAF_COMMON_IMAGE=nexus3.edgexfoundry.org:10003/edgex-taf-common${USE_ARM64}:latest
COMPOSE_IMAGE=nexus3.edgexfoundry.org:10003/edgex-devops/edgex-compose${USE_ARM64}:latest

```

Can I test TAF on Risc-V architecture systems? If yes, how do I define the risc-v architecture in the shell script? 

Any help would be appreciated.

**Comment by cloudxxx8** [comment]
We don't have the Risc-V image so far.  You can build the images on your platform.

**Reply by neerajasjawali** [reply]
> We don't have the Risc-V image so far. You can build the images on your platform.

@cloudxxx8, How do I define Risc-V architecture in the run-tests.sh script? Like riscv_64 or riscv64, what's the correct way to name it?

**Reply by bnevis-i** [reply]
> > We don't have the Risc-V image so far. You can build the images on your platform.
> 
> @cloudxxx8, How do I define Risc-V architecture in the run-tests.sh script? Like riscv_64 or riscv64, what's the correct way to name it?

You want the output of `uname -m`

**Reply by neerajasjawali** [reply]
> > > We don't have the Risc-V image so far. You can build the images on your platform.
> > 
> > 
> > @cloudxxx8, How do I define Risc-V architecture in the run-tests.sh script? Like riscv_64 or riscv64, what's the correct way to name it?
> 
> You want the output of `uname -m`

Thank you @bnevis-i

**Reply by neerajasjawali** [reply]
Within the shell scripts the `USE_ARCH` is used. Do I need to change the name of architecture from arm or x86 to riscv64?

**Reply by cloudxxx8** [reply]
The change in the test.sh is very minor.  EdgeX doesn't provide Risc-V images for TAF and services currently.
There will be lots of changes to do.

**Reply by neerajasjawali** [reply]
Hi @cloudxxx8 , we have already built the risc-v images for all the services. Please let me know how can I modify the below variables to be used in my risc-v environment?

```
TAF_COMMON_IMAGE=nexus3.edgexfoundry.org:10003/edgex-taf-common${USE_ARM64}:latest
COMPOSE_IMAGE=nexus3.edgexfoundry.org:10003/edgex-devops/edgex-compose${USE_ARM64}:latest

```

**Reply by farshidtz** [reply]
Those are simply Docker image names. If your built images are available locally or on Dockerhub: `image_name:tag`. If on a different registry, as with EdgeX's nexus, then: `host:port/image_name:tag`

**Reply by neerajasjawali** [reply]
> Those are simply Docker image names. If your built images are available locally or on Dockerhub: `image_name:tag`. If on a different registry, as with EdgeX's nexus, then: `host:port/image_name:tag`

Hi, since we are doing it for Risc-V, this particular compose_Image do we have to built it or is it already available?   
`COMPOSE_IMAGE=nexus3.edgexfoundry.org:10003/edgex-devops/edgex-compose${USE_ARM64}:latest`
